### PR TITLE
Use updated `sign` method

### PR DESCRIPTION
**Pull request by kchulj**
Transition from the `deprecated` `sign` to the new `sign`.

Also includes a commit which filters `Amount`s less than `5,000,000`.
Smaller coin units should be filtered, so that there is enough amount in the output for the fee deduction.

Part of #26

### CODE REVIEW

**Comment by kchulj** [comment]
This is draft until it is merged step-by-step with bosagora/agora#2370.

File: source/faucet/main.d [hunk 1]
```diff
@@ -441,8 +443,8 @@ public class Faucet : FaucetAPI
                 leftover.sub(new_utxo.value.output.value);
             }
 
-            Transaction tx = txb
-                .sign(OutputType.Payment, 0, &this.keyUnlocker);
+            Transaction tx = txb.unlockSigner(&this.keyUnlocker)
+                                .sign(OutputType.Payment, 0);
```
Comment (hewison-chris) [inline source/faucet/main.d:447]: Extra whitespace.

File: source/faucet/main.d [hunk 2]
```diff
@@ -274,7 +274,8 @@ public class Faucet : FaucetAPI
         auto builder = TxBuilder(
             secret_keys.byKey().drop(uniform(0, secret_keys.length, rndGen)).front());
         builder.attach(utxo_rng);
-        return builder.sign(OutputType.Payment, 0, &this.keyUnlocker);
+        return builder.unlockSigner(&this.keyUnlocker)
+                      .sign(OutputType.Payment, 0);
```
Comment (hewison-chris) [inline source/faucet/main.d:278]: Extra whitespace.

**Review by hewison-chris — APPROVED** [review]
LGTM just some extra whitespace.

File: source/faucet/main.d (hunk 2 above)
Comment (kchulj) [reply source/faucet/main.d:278]: Where is the whitespace?

File: source/faucet/main.d (hunk 1 above)
Comment (kchulj) [reply source/faucet/main.d]: Updated.

File: source/faucet/main.d [hunk 3]
```diff
@@ -238,14 +242,14 @@ public class Faucet : FaucetAPI
...
+            .filter!(tup => tup.value.output.value >= minInputValuePerOutput * count)
             .map!(tup => TxBuilder(tup.value.output, tup.key))
-            .map!(txb => txb.split(
+            .map!(txb => txb.unlockSigner(&this.keyUnlocker).split(
                     secret_keys.byKey() // AA keys are addresses
                     .cycle()    // cycle the range of keys as needed
                     .drop(uniform(0, count, rndGen))    // start at some random position
                     .take(count))
-                .sign(OutputType.Payment, 0, &this.keyUnlocker));
+                .sign(OutputType.Payment, 0));
```
Comment (hewison-chris) [inline source/faucet/main.d:252]:
```suggestion
                .sign());
```

File: source/faucet/main.d [hunk 4]
```diff
@@ -270,7 +274,8 @@ public class Faucet : FaucetAPI
         auto builder = TxBuilder(
             secret_keys.byKey().drop(uniform(0, secret_keys.length, rndGen)).front());
         builder.attach(utxo_rng);
-        return builder.sign(OutputType.Payment, 0, &this.keyUnlocker);
+        return builder.unlockSigner(&this.keyUnlocker)
+            .sign(OutputType.Payment, 0);
```
Comment (hewison-chris) [inline source/faucet/main.d:278]:
```suggestion
            .sign();
```

File: source/faucet/main.d [hunk 5]
```diff
@@ -406,8 +412,9 @@ public class Faucet : FaucetAPI
 
         if (leftover <= first_utxo.value.output.value)
         {
-            Transaction tx = txb.draw(leftover, [pubkey])
-                .sign(OutputType.Payment, 0, &this.keyUnlocker);
+            Transaction tx = txb.unlockSigner(&this.keyUnlocker)
+                .draw(leftover, [pubkey])
+                .sign(OutputType.Payment, 0);
```
Comment (hewison-chris) [inline source/faucet/main.d:417]:
```suggestion
                .sign();
```

File: source/faucet/main.d [hunk 6]
```diff
@@ -436,8 +443,8 @@ public class Faucet : FaucetAPI
                 leftover.sub(new_utxo.value.output.value);
             }
 
-            Transaction tx = txb
-                .sign(OutputType.Payment, 0, &this.keyUnlocker);
+            Transaction tx = txb.unlockSigner(&this.keyUnlocker)
+                .sign(OutputType.Payment, 0);
```
Comment (hewison-chris) [inline source/faucet/main.d:447]:
```suggestion
                .sign();
```

**Review by hewison-chris — CHANGES_REQUESTED** [review]
LGTM but default params for sign can be used.

File: source/faucet/main.d (hunk 2 above)
Comment (hewison-chris) [reply source/faucet/main.d]: Before the `.sign`. Should be like line 274.

**Comment by kchulj** [comment]
> LGTM but default params for sign can be used.

Updated

**Review by hewison-chris — APPROVED** [review]
LGTM

**Review by hewison-chris — APPROVED** [review]
LGTM

**Review by Geod24 — APPROVED** [review]

File: source/faucet/main.d [hunk 7]
```diff
@@ -238,14 +242,14 @@ public class Faucet : FaucetAPI
         static assert (isInputRange!UR);
 
         return utxo_rng
-            .filter!(tup => tup.value.output.value >= Amount(count))
+            .filter!(tup => tup.value.output.value >= minInputValuePerOutput * count)
             .map!(tup => TxBuilder(tup.value.output, tup.key))
-            .map!(txb => txb.split(
+            .map!(txb => txb.unlockSigner(&this.keyUnlocker).split(
```
Comment (Geod24) [inline source/faucet/main.d:247]: I think we should be a bit smarter here (attaching to multiple UTXOs) but for now this works.

**Pull request merged**